src/main/java/io/vavr/collection/Iterator.java

```diff
@@ -742,39 +742,48 @@ public BigDecimal next() {
         } else if (Integer.signum(step) == Integer.signum(from - toInclusive)) {
             return empty();
         } else {
-            final int end = toInclusive - step;
             if (step > 0) {
                 return new Iterator<Integer>() {
-                    int i = from - step;
+                    int next = from;
+                    boolean overflow = false;
 
                     @Override
                     public boolean hasNext() {
-                        return i <= end;
+                        return !overflow && next <= toInclusive;
                     }
 
                     @Override
                     public Integer next() {
                         if (!hasNext()) {
                             throw new NoSuchElementException();
                         }
-                        return i += step;
+                        int curr = next;
+                        int r = curr + step;
+                        overflow = ((curr ^ r) & (step ^ r)) < 0;
+                        next = r;
+                        return curr;
```

Comment:
Well done, Mincong, your solutions (not only this one) are well thought through!

I would love if I may invite you to join the group of official VAVR committers! It is meant as kind of achievement :)

Reply:
Thank you @danieldietrich for your invitation. It's really a pleasure to hear that! I would love to contribute more in the future :)